Code/Tools/FBuild/FBuildCore/Graph/NodeGraph.cpp

```diff
@@ -1413,90 +1413,75 @@ Node * NodeGraph::FindNodeInternal( const AString & fullPath ) const
     return nullptr;
 }
 
+// Comparaison operator for NodeWithDistance
+//------------------------------------------------------------------------------
+bool operator < ( const NodeGraph::NodeWithDistance & lhs, const NodeGraph::NodeWithDistance & rhs  )
+{
+    return ( lhs.m_Distance == rhs.m_Distance
+        ? ( lhs.m_Node->GetName() < rhs.m_Node->GetName() )
+        : ( lhs.m_Distance < rhs.m_Distance )
+    );
+}
+
 // FindNearestNodesInternal
 //------------------------------------------------------------------------------
-void NodeGraph::FindNearestNodesInternal( const AString & fullPath, Array< NodeWithDistance > & nodes, const uint32_t maxDistance ) const
+void NodeGraph::FindNearestNodesInternal( const AString & fullPath, Array< NodeWithDistance > & nodes ) const
 {
     ASSERT( Thread::IsMainThread() );
     ASSERT( nodes.IsEmpty() );
     ASSERT( false == nodes.IsAtCapacity() );
 
-    if ( fullPath.IsEmpty() )
-    {
-        return;
-    }
+    const uint32_t SIMILAR_ENOUGH = 9; // empirical number
 
-    uint32_t worstMinDistance = fullPath.GetLength() + 1;
+    Array< NodeWithDistance > tmp;
 
-    for ( size_t i = 0 ; i < NODEMAP_TABLE_SIZE ; i++ )
+    const size_t totalNodes = GetNodeCount();
+    for ( size_t i = 0 ; i < totalNodes ; ++i )
     {
-        for ( Node * node = m_NodeMap[i] ; nullptr != node ; node = node->m_Next )
+        Node * node = GetNodeByIndex( i );
+        bool displayName = false;
+        switch ( node->GetType() )
         {
```

Comment:
What's the thinking behind limiting to certain node types? While the ones you limited the search to are the most useful, I could see the others still being useful. Was is just to de-clutter the output or was it for performance reasons?